WordPress/gutenberg
# Site Title: Add border block support

## What?

Add border block support to the `Site Title` block.

Part of https://github.com/WordPress/gutenberg/issues/43247

## Why?
`Site Title` block is missing border support.

## How?

Adds the border block support in block.json

## Testing Instructions

- Go to Global Styles setting ( under appearance > editor > styles > edit styles > blocks )
- Make sure that `Site Title` block's border is configurable via Global Styles
- Verify that Global Styles are applied correctly in the editor and frontend
- Edit template/page, Add `Site Title` block and Apply the border styles
- Verify that block borders take precedence over global styles
- Verify that block borders display correctly in both the editor and frontend

## Screenshots or screencast 
In backend:
<img width="1246" alt="Screenshot at Jul 16 22-30-29" src="https://github.com/user-attachments/assets/25067629-7ea6-4ec7-82b4-89e54aeb2b55">
In frontend: 
<img width="815" alt="Screenshot at Jul 16 22-30-50" src="https://github.com/user-attachments/assets/2fbbf504-6acd-43c1-8391-ff0751454375">
